### PR TITLE
Uyenphamh-week7: Develop test scenarios using features file for Opencart web application 

### PR DESCRIPTION
**Pull request by Uyenphamh**
## Exercise

* [Exercise-7](https://github.com/NT-Line-team/Behave_test/issues/9)

## Overview
Develop test scenarios using features file of below function:
Login
Register
Forgot password
Search a product
Buy a product and checkout
Add review for product

## Verification Steps
Open and verify the scenario for 6 features

### Docs
1. add_review.feature
2. buy_product.feature
3. forgot_password.feature
4. login.feature
5. register.feature
6. search_product.feature

### CODE REVIEW

**Comment by Uyenphamh** [comment]
Hi anh @hungphamviet  and anh @duynguyenx 

Please help me to review these feature files.

Thank you,
Uyen Pham

File: exercises/uyenphamh/add_review.feature [hunk 1]
```diff
@@ -0,0 +1,18 @@
+Feature: Add review product
```
Comment (duynguyenx) [inline exercises/uyenphamh/add_review.feature]: pls put all features file to `/features/your_name/example.feature`

File: exercises/uyenphamh/add_review.feature [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+
```
Comment (duynguyenx) [inline exercises/uyenphamh/add_review.feature]: Missing user role and what he/she want to check in this feature. Pls see below example:
`As a standard user, I want to add a review for product that i bought`, pls also correct other features.

File: exercises/uyenphamh/add_review.feature [hunk 3]
```diff
@@ -0,0 +1,18 @@
+Feature: Add review product
+		
+	Scenario: Write review product
```
Comment (duynguyenx) [inline exercises/uyenphamh/add_review.feature]: Pls add tags for both feature and test cases in feature

File: exercises/uyenphamh/add_review.feature [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+		And I click on the result item search
+		And I click write a review link
+		When I type <name> in your name field
+		And I type <review> in your review field
+		And I choose <rate> in rating
+		And I click continue button
+		Then I see <validation_message>
+
+	Examples:
+	| name | review | rate | validation_message|
```
Comment (duynguyenx) [inline exercises/uyenphamh/add_review.feature]: Pls re-format the context table to make it more readable.

File: exercises/uyenphamh/add_review.feature [hunk 5]
```diff
@@ -0,0 +1,18 @@
+Feature: Add review product
+		
+	Scenario: Write review product
+		Given I search “iPhone”
```
Comment (duynguyenx) [inline exercises/uyenphamh/add_review.feature]: Pls correct your test steps using below suggestion
1. Given: the test case pre-condition.
2. When: step/action.
3. Then: things you want to check.

**Comment by Uyenphamh** [comment]
Hi anh @duynguyenx 

Thank you for your comment.
I've updated pls review it.

Thank you

**Pull request merged**